contracts/BaseLaunchpeg.sol

```diff
@@ -360,6 +431,76 @@ abstract contract BaseLaunchpeg is
         emit DevMint(msg.sender, _quantity);
     }
 
+    /// @dev Should only be called in the pre-mint phase
+    /// @param _quantity Quantity of NFTS to mint
+    function _preMint(uint256 _quantity) internal {
+        if (_quantity == 0) {
+            revert Launchpeg__InvalidQuantity();
+        }
+        if (_quantity > allowlist[msg.sender]) {
+            revert Launchpeg__NotEligibleForAllowlistMint();
+        }
+        if (
+            totalSupply() + _quantity > collectionSize ||
+            amountMintedDuringAllowlist + _quantity > amountForAllowlist
+        ) {
+            revert Launchpeg__MaxSupplyReached();
+        }
+        allowlist[msg.sender] -= _quantity;
+        uint256 price = _preMintPrice();
+        uint256 totalCost = price * _quantity;
+        numberPreMinted[msg.sender] += _quantity;
+        amountMintedDuringPreMint += _quantity;
+        amountMintedDuringAllowlist += _quantity;
+        preMintQueue.push(
+            PreMintData({sender: msg.sender, quantity: _quantity})
+        );
+        emit PreMint(msg.sender, _quantity, price);
+        _refundIfOver(totalCost);
+    }
+
+    /// @dev Should only be called in the allowlist and public sale phases.
+    /// @param _maxQuantity Max quantity of NFTs to mint
+    function _batchMint(uint256 _maxQuantity) internal {
+        if (_maxQuantity == 0) {
+            revert Launchpeg__InvalidQuantity();
+        }
+        if (amountMintedDuringPreMint == amountBatchMinted) {
+            revert Launchpeg__MaxSupplyForBatchMintReached();
+        }
+        uint256 remQuantity = _maxQuantity;
+        uint256 price = _preMintPrice();
+        address sender;
+        uint256 quantity;
+        uint256 i = preMintQueueIdx;
+        uint256 length = preMintQueue.length;
+        while (i < length && remQuantity > 0) {
+            PreMintData memory data = preMintQueue[i];
+            sender = data.sender;
+            if (data.quantity > remQuantity) {
+                quantity = remQuantity;
+                preMintQueue[i].quantity -= quantity;
+            } else {
+                quantity = data.quantity;
+                delete preMintQueue[i];
+                i++;
```

Comment:
I don't think we cover these lines in the tests?

Let's say Alice pre-minted 10 and Bob pre-minted 5

We call `batchMint(10)` then `batchMint(5)`, are we sure this works?

Reply:
updated the test [here](https://github.com/traderjoe-xyz/launchpeg/pull/41/files#diff-1a1e34669e6ec9cb87b3945e3eb6e8e0bbdfd07f18a4f5eda9ac771f0106e263R344)